### PR TITLE
polkadot-node-subsystems: `ChainApiBackend` added + `polkadot-debug` image version fixed

### PR DESCRIPTION
**Pull request by michalkucharczyk**
The out-dated version (bad tag) of [polkadot image](https://github.com/paritytech/polkadot-sdk/blob/ede4a362622dfa69eb167eaa876246b1289f4b41/.gitlab/pipeline/zombienet/cumulus.yml#L31) ([docker info](https://hub.docker.com/layers/paritypr/polkadot-debug/master/images/sha256:adb1658052cf671b50c90d5cece5c7a131efa1a95978249bd5cb85a5ad654f7a?context=explore))  was used. This PR fixes this.

See also: https://github.com/paritytech/polkadot-sdk/pull/2411#issuecomment-1822632724

### CODE REVIEW

**Review by pepoviola — APPROVED** [review]

**Review by skunert — APPROVED** [review]

**Review by sergejparity — APPROVED** [review]

**Review by AndWeHaveAPlan — APPROVED** [review]

**Review by Ank4n — APPROVED** [review]

**Comment by skunert** [comment]
Sadly while the wrong image was used, collator RPC functionality broke. Since tests were running against an old version of polkadot, they remained green and it went unnoticed.

To repair this functionality we need the `ChainApiSubsystem` in the collator again, so I added it. It was expecting a `HeaderBackend` implementation which we can not provide on the collator. I introduced an async wrapper, `ChainApiBackend`. Since this now touches the chain api subsystem, I would like to have a review from someone from the parachains team.

File: polkadot/node/subsystem-types/src/runtime_client.rs [hunk 1]
```diff
@@ -18,17 +18,70 @@ use async_trait::async_trait;
...
+/// `ChainApiSubsystem`.
+// This trait was introduced to suit the needs of collators. Depending on their operating mode, they
+// might not have a client of the relay chain that can supply a synchronous HeaderBackend
+// implementation.
+#[async_trait]
+pub trait ChainApiBackend: Send + Sync {
+	/// Get block header. Returns `None` if block is not found.
+	async fn header(&self, hash: Hash) -> sp_blockchain::Result<Option<Header>>;
+	/// Get blockchain info.
+	async fn info(&self) -> sp_blockchain::Result<Info<Block>>;
```
Comment (ordian) [inline polkadot/node/subsystem-types/src/runtime_client.rs:46]: It seems at the present, we don't need the full info to satisfy the request, just the latest finalized number. 
If we return just that, the implementation for collator side can be simplified. That being said, collator should not be (transitively) calling this method AFAIK (at least minimal relay chain node), so it should be fine as is.

**Review by ordian — APPROVED** [review]
The introduced `ChainApiBacked` async abstraction looks good to me (modulo comment). 

We were planning to add chain api subsystem to the (minimal) collator anyway in #1644.

File: cumulus/client/relay-chain-minimal-node/src/blockchain_rpc_client.rs [hunk 2]
```diff
@@ -54,6 +56,65 @@ impl BlockChainRpcClient {
...
+	async fn number(
+		&self,
+		hash: <Block as BlockT>::Hash,
+	) -> sp_blockchain::Result<Option<<<Block as BlockT>::Header as HeaderT>::Number>> {
+		let result = self
+			.rpc_client
+			.chain_get_header(Some(hash))
+			.await?
+			.map(|maybe_header| maybe_header.number);
+		Ok(result)
```
Comment (michalkucharczyk) [inline cumulus/client/relay-chain-minimal-node/src/blockchain_rpc_client.rs:107]:
```suggestion
		Ok(self
			.rpc_client
			.chain_get_header(Some(hash))
			.await?
			.map(|maybe_header| maybe_header.number))
```

**Comment by michalkucharczyk** [comment]
lgtm

**Pull request merged**